ios/RN/RNCamera.m

```diff
@@ -222,28 +222,34 @@ - (void)updateAutoFocusPointOfInterest
 {
     AVCaptureDevice *device = [self.videoCaptureDeviceInput device];
     NSError *error = nil;
-
+    
     if (![device lockForConfiguration:&error]) {
         if (error) {
             RCTLogError(@"%s: %@", __func__, error);
         }
         return;
     }
-
+    
     if ([self.autoFocusPointOfInterest objectForKey:@"x"] && [self.autoFocusPointOfInterest objectForKey:@"y"]) {
         float xValue = [self.autoFocusPointOfInterest[@"x"] floatValue];
         float yValue = [self.autoFocusPointOfInterest[@"y"] floatValue];
         if ([device isFocusPointOfInterestSupported] && [device isFocusModeSupported:AVCaptureFocusModeContinuousAutoFocus]) {
-
+            
             CGPoint autofocusPoint = CGPointMake(xValue, yValue);
             [device setFocusPointOfInterest:autofocusPoint];
             [device setFocusMode:AVCaptureFocusModeContinuousAutoFocus];
-          }
+            
+            // Manual selection of AFPoI also resets the manual exposure, if there was any.
+            if(self.exposure >= 0 && self.exposure <= 1){
+                self.exposure = -1;
+                [device setExposureMode:AVCaptureExposureModeContinuousAutoExposure];
```

Comment:
Why do we change exposure mode here without users directive?

Reply:
This reverts the exposure mode to the default behavoir. The exposure mode gets only changed by user if the exposure-value gets changed. As soon as the user does a tap-to-focus the exposure mode gets reset to its original state.

Reply:
If it is set, can we not leave the exposure mode at the level they set?  Why do we have to reset it if the device is not reseting it? Or is the device reseting it, so this code is just to be true to what the device is automatically doing.

Asking because i have a PR to set auto-exposure point of interest to the same point, when auto focus point of interest is set.

Reply:
This code mimics the stock iOS camera-app:

1. App starts with `AVCaptureExposureModeContinuousAutoExposure`
2. User starts to manually adapt the ISO value by increasing/reducing it, which as a side effect also changes the exposure-mode to `AVCaptureExposureDurationCurrent` else the changed ISO-value would be changed again by the AVFoundation-kit
3. The user does a tap-to-focus, which in the stock app reverts to `AVCaptureExposureModeContinuousAutoExposure` and adapts the ISO based on the focus point of interest. This behavior is copied in RNCamera by this code.

Reply:
Oh thank you very much for that explanation. Do you know if the stock app also sets the exposure point of interest, when the focus point is set?

Reply:
I very much think so. At least when trying out against a dark surface on one side a light surface on the other side, the exposure also gets corrected based on the focus point, which is where I tapped.

Reply:
I think so too. Please let me think for a bit about if we should add setting EPoI here as well.

Reply:
Hey @CaptainQuark I'm thinking that using `autoFocusPointOfInterest` does not necessarily mean they want "tap to zoom", so maybe we shouldn't replicate that. I also will have to adjust my PR #2423 - what do you think?

If the brightness is locked, even when point of itnerest is set, it should maintain that, unless they use a control to adjust the brightness, or switch to auto brightness. (or auto focus)

Reply:
Hey @CaptainQuark I'm thinking that using `autoFocusPointOfInterest` does not necessarily mean they want "tap to zoom", so maybe we shouldn't replicate that. I also will have to adjust my PR #2423 - what do you think?

If the brightness is locked, even when point of itnerest is set, it should maintain that, unless they use a control to adjust the brightness, or switch to auto brightness. (or auto focus)

Reply:
Hey Yasir,

my two cents:
- this PR aims at harmonizing the exposure control on both iOS & Android. RNCamera suffers more than other libs from fragmentation, as the underlying mechanisms are, in certain areas, just very different. By providing a single, simple interface for exposure control, we add new functionality and help reduce this fragmentation in code a little bit
- your PR only tackles the iOS-side of things and imho doesn't provide a really useful extension of RNCamera's skills, as an exposure-point-of-interest doesn't event exist on Android
- keeping the end-users in mind, tap-to-focus works on both iOS & Android very similar, so we should adapt this already known behavior and not try to complicate things further
- I don't see a need for your locked exposure-control-point-of-interest

Although I might sound harsh here, I think clearly pointing out the advantages/disadvantages of both PRs helps everybody. RNCamera benefits most from a unified interface for all platforms and functions that a majority of developers will use.

Reply:
Oh Android device resets exposure when a focus point is selected? And iOS device does not reset exposure when focus point is selected? This is why you reset the exposure on iOS? It's not possible to get Android to maintain the exposure level? Is this true? Getting a unified interface makes sense, but then we should pick either (1) reset ios exposure on auto focus or (2) keep android exposure constant on auto focus.

The auto-exposure use case is there, but I'll put that discussion in the other PR, I don't want to take this PR off topic.

Reply:
@CaptainQuark - what do you say? Should we respect the exposure prop? Or should we keep PR the same and continue to reset exposure despite prop what prop says? @sibelius and others, we can use your input here as well.